src/main/Extensions/EmptyTrash/EmptyTrashExtension.ts

```diff
@@ -0,0 +1,116 @@
+import type { AssetPathResolver } from "@Core/AssetPathResolver";
+import type { Extension } from "@Core/Extension";
+import type { FluentIcon, SearchResultItem } from "@common/Core";
+import type { Image } from "@common/Core/Image";
+
+type Settings = {
+    confirmBeforeEmpty: boolean;
+};
+
+export class EmptyTrashExtension implements Extension {
+    public readonly id = "EmptyTrash";
+    public readonly name = "Empty Trash";
+
+    public readonly nameTranslation = {
+        key: "Empty Trash",
+        namespace: "extension[EmptyTrash]",
+    };
+
+    private readonly defaultSettings: Settings = {
+        confirmBeforeEmpty: true,
+    };
+
+    public readonly author = {
+        name: "Ömer Duran",
+        githubUserName: "omerdduran",
+    };
+
+    public constructor(private readonly assetPathResolver: AssetPathResolver) {}
+
+    public async getSearchResultItems(): Promise<SearchResultItem[]> {
+        return [
+            {
+                name: "Empty Trash",
+                description: "Empty system trash/recycle bin",
+                descriptionTranslation: {
+                    key: "emptyTrashDescription",
+                    namespace: "extension[EmptyTrash]",
+                },
+                id: "emptyTrash:action",
+                image: this.getImage(),
+                defaultAction: {
+                    description: "Empty trash",
+                    descriptionTranslation: {
+                        key: "emptyTrashAction",
+                        namespace: "extension[EmptyTrash]",
+                    },
+                    handlerId: "EmptyTrash",
+                    argument: "empty",
+                    fluentIcon: "DeleteDismissRegular" as FluentIcon,
+                    hideWindowAfterInvocation: true,
+                    requiresConfirmation: true,
+                },
+            },
+        ];
+    }
+
+    public getInstantSearchResultItems(searchTerm: string): SearchResultItem[] {
+        if (!searchTerm.toLowerCase().includes("trash")) {
+            return [];
+        }
+
+        return [
+            {
+                name: "Empty Trash",
+                description: "Empty system trash/recycle bin",
+                descriptionTranslation: {
+                    key: "emptyTrashDescription",
+                    namespace: "extension[EmptyTrash]",
+                },
+                id: "emptyTrash:action",
+                image: this.getImage(),
+                defaultAction: {
+                    description: "Empty trash",
+                    descriptionTranslation: {
+                        key: "emptyTrashAction",
+                        namespace: "extension[EmptyTrash]",
+                    },
+                    handlerId: "EmptyTrash",
+                    argument: "empty",
+                    fluentIcon: "DeleteDismissRegular" as FluentIcon,
```

Comment:
If you update your branch with the latest `main` branch then this type casting is not needed.

```suggestion
                    fluentIcon: "DeleteDismissRegular",
```